overlord/fdestate/fdemgr.go

```diff
@@ -132,19 +132,41 @@ func (m *FDEManager) ReloadModeenv() error {
 	return initModeFromModeenv(m)
 }
 
-func (m *FDEManager) resealKeyForBootChains(unlocker boot.Unlocker, method device.SealingMethod, rootdir string, params *boot.ResealKeyForBootChainsParams, expectReseal bool) error {
-	doUpdate := func(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error {
-		if unlocker != nil {
-			m.state.Lock()
-			defer m.state.Unlock()
-		}
-		return updateParameters(m.state, role, containerRole, bootModes, models, tpmPCRProfile)
+type unlockedStateManager struct {
+	*FDEManager
+	unlocker boot.Unlocker
+	relock   func()
+}
+
+func (m *unlockedStateManager) Update(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error {
+	return m.UpdateParameters(role, containerRole, bootModes, models, tpmPCRProfile)
+}
+
+func (m *unlockedStateManager) Get(role string, containerRole string) (bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte, err error) {
+	return m.GetParameters(role, containerRole)
+}
+
+func (m *unlockedStateManager) Unlock() {
```

Comment:
it feels like returning the relock function or a nop function would be a bit simpler that needing to have a relock member?